app/bundles/StageBundle/Tests/Functional/Controller/StageController.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Mautic\StageBundle\Tests\Functional\Controller;
+
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use Mautic\LeadBundle\Entity\Lead;
+use Mautic\LeadBundle\Model\LeadModel;
+use Mautic\StageBundle\Cache\StageCountCache;
+use Mautic\StageBundle\Entity\Stage;
+use Mautic\StageBundle\Model\StageModel;
+
+class StageController extends MauticMysqlTestCase
+{
+    private StageCountCache $stageCountCache;
+
+    private StageModel $stageModel;
+
+    private LeadModel $leadModel;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->stageCountCache = self::getContainer()->get(StageCountCache::class);
+        $this->stageModel      = self::getContainer()->get(StageModel::class);
+        $this->leadModel       = self::getContainer()->get(LeadModel::class);
+    }
+
+    public function testGetStageContactCount(): void
+    {
+        $contact = new Lead();
+        $contact->setEmail('john@doe.com');
+        $this->leadModel->saveEntity($contact);
+
+        $stage = new Stage();
+        $stage->setName('Stage1');
+        $stage->setWeight(1);
+        $this->stageModel->saveEntity($stage);
+
+        $this->assertEquals(0, $this->stageCountCache->getStageContactCount($stage->getId()));
```

Comment:
@kuzmany please replace these direct calls with request calls to your new controller action. It will test the whole feature, not just a part of it.